dao/src/main/java/com/epam/esm/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,87 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.OrderDao;
+import com.epam.esm.dao.exception.EntityNotInDBException;
+import com.epam.esm.domain.entity.Order;
+import org.springframework.stereotype.Repository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import java.util.List;
+import java.util.Optional;
+
+import static com.epam.esm.dao.util.ParameterValue.INT_VALUE_OFFSET;
+
+@Repository
+public class OrderDaoImpl implements OrderDao {
+
+    EntityManager entityManager;
+
+    private static final String ATTRIBUTE_USER_ID = "userId";
+
+    public OrderDaoImpl(EntityManager entityManager) {
+        this.entityManager = entityManager;
+    }
+
+    @Override
+    public List<Order> readAll(int page, int size) {

```

Comment:
This method, readById and getEntitiesCount are the same in all DAO classes that makes them candidates to extracting to common place.

Reply:
thanks for advice. methods extracted to common implementation class